File: HW4/Ball.cpp
#include"Ball.h"

Ball::Ball() {
	radius = 1;
}
Ball::Ball(const double& radius) {
	setRadius(radius);
}
Ball::Ball(double radius, const string& color, const bool& filled) {
	setRadius(radius);
	setColor(color);
	setFilled(filled);
}

void Ball::setRadius(double newRadius) {
	radius = newRadius;
}
double Ball::getRadius() const {
	return radius;
}

double Ball::getVolume()  {
	return 4 * 3.14159 * pow(radius, 3) / 3;
}
double Ball::getArea()  {
	return 4 * 3.14159 * pow(radius, 2);
}

string Ball::toString()  {
	stringstream ss;
	ss << "ball, " << radius;
	return ss.str();
}

File: Python/第15章/15-2/Ball.py
from GeometricObject import GeometricObject

class Ball(GeometricObject):
    def __init__(self, radius, color):
        self.__radius = radius
        super().setColor(color)

    def setRadius(self, radius):
        self.__radius = radius
    def getRadius(self):
        return self.__radius

    def getVolume(self):
        return 4 * 3.14159 * self.__radius ** 3 / 3
    def getArea(self):
        return 4 * 3.14159 * self.__radius ** 2